### PR TITLE
57 connect audio service to grpc

### PR DESCRIPTION
**Pull request by glannuzel**
gRPC server should be available for the client.
Proto functions are all declared but not implemented.

### CODE REVIEW

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py [hunk 1]
```diff
@@ -15,92 +29,104 @@ class ReachyGRPCAudioSDKServicer:
...
-
-    def stop(self) -> None:
-        # note: not sure it is working. Need to double check ROS package
-        self.node.get_logger().info(f"Stop playing")
-        self.soundhandle.stopAll()
-
-    def start_capture(self, filename: str) -> bool:
-        if not pathlib.Path(filename).parent.absolute().exists():
-            self.node.get_logger().error(f"Path does not exist {filename}")
+    def GetAllMicrophone(
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py:42]: We'll we have multiple mic and speakers? I'd use the default ones we configure with the robot (antennas for the mics and respeakers for the HP)

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py [hunk 2]
```diff
@@ -15,92 +29,104 @@ class ReachyGRPCAudioSDKServicer:
...
+
+        self.node.get_logger().info("Reachy GRPC Audio SDK Servicer initialized.")
 
-        self.node.get_logger().info("Playing say-beep at volume 0.3.")
-        self.soundhandle.playWave("say-beep.wav", volume=0.3)
+    def register_to_server(self, server: grpc.Server):
+        self.node.get_logger().info("Registering 'SoundServiceServicer' to server.")
+        add_SoundServiceServicer_to_server(self, server)
 
     def say_text(self, text: str) -> None:
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py:38]: Should we add a function in the API for this?
We can use this lib for now and if we think to keep this feature we'll choose a better voice

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py [hunk 3]
```diff
@@ -15,92 +29,104 @@ class ReachyGRPCAudioSDKServicer:
...
+        return Empty()
+
+    def ChangeVolume(
+        self, request: VolumeRequest, context: grpc.ServicerContext
+    ) -> Empty:
+        self._volume = request.id
+        return Empty()
+
+    def PlaySound(self, request: SoundId, context: grpc.ServicerContext) -> Empty:
+        self.node.get_logger().info(f"Playing {request.id}")
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: Should we check that the path exists?

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py [hunk 4]
```diff
@@ -15,92 +29,104 @@ class ReachyGRPCAudioSDKServicer:
...
+    def GetAllSpeaker(
+        self, request: Empty, context: grpc.ServicerContext
+    ) -> ListOfSpeaker:
+        return ListOfSpeaker(
+            speaker_info=[Speaker(id=ComponentId(id=1, name="speaker_1"))]
+        )
+
+    def StartRecording(
+        self, request: RecordingRequest, context: grpc.ServicerContext
+    ) -> Empty:
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: Return true or false if the recording started or not

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py [hunk 5]
```diff
@@ -15,92 +29,104 @@ class ReachyGRPCAudioSDKServicer:
...
+        return RecordingAck(ack=SoundAck(success=BoolValue(value=True)))
+
+    def TestSpeaker(self, request: ComponentId, context: grpc.ServicerContext) -> Empty:
+        self.soundhandle.playWave("say-beep.wav", volume=self._volume)
+        return Empty()
+
+    def ChangeVolume(
+        self, request: VolumeRequest, context: grpc.ServicerContext
+    ) -> Empty:
+        self._volume = request.id
```
Comment (FabienDanieau) [inline reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: not sure is request.id is good change it not

Comment (FabienDanieau) [reply reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: need to check if it is in [0,1.0]. Here or checked on the user side?

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py (hunk 1 above)
Comment (simheo) [reply reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py:42]: We considered on the client side that we only have one mic and speaker. I don't know if in the future we might want to have multiple mic and / or speakers

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py (hunk 3 above)
Comment (simheo) [reply reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: We can on the server side. On the client side we check if the sound to [play is available](https://github.com/pollen-robotics/reachy-v2-sdk/blob/8b67aaf3239b1d10e8247ee89af62a3645de7f25/src/reachy_v2_sdk/audio.py#L53C1-L57C120), I don't know if we need a double check.

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py (hunk 4 above)
Comment (simheo) [reply reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py]: If we can catch this on the system let's do that yes!

File: reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py (hunk 2 above)
Comment (FabienDanieau) [reply reachy_sdk_server/reachy_sdk_server/grpc_server/audio_server.py:38]: will be addressed here https://github.com/pollen-robotics/reachy-v2-sdk/issues/123

**Comment by FabienDanieau** [comment]
To be tested with this docker image https://github.com/pollen-robotics/docker_sdk_server/tree/3-add-audio-support

**Comment by FabienDanieau** [comment]
Will close #56 (already merged in #57)

**Pull request merged**